### PR TITLE
feat: introduce code owners file

### PR DESCRIPTION
**Pull request by winged**
Let's see how this works out. This here is more of an experiment than
a production setting. If it gains traction, we'll add it to more projects
in the future

### CODE REVIEW

File: CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Code owners for the Timed backend. We include our backend dev team here.
+# Since this is a split project (backend/frontend) it's rather simple
+* @adfinis-sygroup/dev/backend
```
Comment (sbor23) [inline CODEOWNERS:3]: Does the slashed notion have any functional meaning to it? Never seen that before

Comment (winged) [reply CODEOWNERS:3]: That's there to reference (nested) teams

Comment (sbor23) [reply CODEOWNERS:3]: Ok I can't see such a nested example in [the docs](https://docs.github.com/en/enterprise/2.15/user/articles/about-code-owners) but fine with me.

**Review by sbor23 — APPROVED** [review]

File: CODEOWNERS (hunk 1 above)
Comment (winged) [reply CODEOWNERS]: Yeah you're right - nested teams are still referenced via `org/team-name` without nesting. Changed the file accordingly

Comment (winged) [reply CODEOWNERS]: (... and renamed the teams as well for consistency)

**Comment by hairmare** [comment]
I set up repo permissions for the @adfinis-sygroup/dev-backend group but the repo didn't seem to pick up on them yet.

**Pull request merged**